### PR TITLE
initial unit testing CI skeleton

### PR DESCRIPTION
**Pull request by zmk5**
We can use this as an initial skeleton for unit testing. We will need to add some unit test but for now lets see if the current CI interface can unit test PyQt applications.

### CODE REVIEW

**Comment by zmk5** [comment]
PyQt5 is capable of being used in the VMs that they have for the CI interface! We can now start looking into creating unit tests for our UI work 😄

**Comment by zmk5** [comment]
One issue I see is that for some reason this pull request is based off of the `master` branch and not the `develop` branch. If it is an issue I can redo the pull request and hopefully do it from the `develop` branch.

File: .github/workflows/ui_unit_testing.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    # Download PyQt5's dependencies
+    - name: Download PyQt5 Dependencies
+      run: python3 -m pip install setuptools
+
+    # Download our packages dependencies
+    - name: Download PyQt5
+      run: sudo apt install python3-pyqt5
+    
+    - name: Download Other Package Dependencies
+      run: python3 -m pip install numpy pyqtgraph
```
Comment (BrianWoodruff) [inline .github/workflows/ui_unit_testing.yml]: To run tests with serial we'll also need the 'crc16' package

**Comment by BrianWoodruff** [comment]
> One issue I see is that for some reason this pull request is based off of the `master` branch and not the `develop` branch. If it is an issue I can redo the pull request and hopefully do it from the `develop` branch.

Looks like it's applied to develop to me...

**Comment by BrianWoodruff** [comment]
Zahi, thanks for adding a unit test framework!   Looking forward to seeing some example tests so that we can start adding more.

File: .github/workflows/ui_unit_testing.yml (hunk 1 above)
Comment (zmk5) [reply .github/workflows/ui_unit_testing.yml]: Cool, I'll add this too.

**Comment by zmk5** [comment]
Yeah it will be applied to `develop`, but this branch seems to be based on `master` unfortunately...

**Comment by avigupta33** [comment]
Can you merge develop back to your branch and commit it? Would that solve the problem?

**Comment by zmk5** [comment]
Let's try it!

**Comment by zmk5** [comment]
Now the branch resembles `develop` we can now start adding unit tests to our system!

**Review by avigupta33 — APPROVED** [review]

**Pull request merged**